backend/app/services/loop_items/external_provider.py

```
@@ -0,0 +1,579 @@
+    ) -> dict[str, object]:
+        access = require_cloud_project_role(
+            db, project_id, user_id, BaseRole.RestrictedAnalyst
+        )
+        project = access.project
+        self._require_external(project)
+        if not access.is_public_visitor and not has_permission(
+            access.role, BaseRole.Reporter
+        ):
+            raise HTTPException(status.HTTP_403_FORBIDDEN, "Insufficient permission")
+        labels = self._labels_for_write(
+            values.tags + [f"{CREATOR_PREFIX}{user_id}:{self._safe_name(user_name)}"],
```

Ruff (0.15.21)

[warning] Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)